core/DbHelper.php

```diff
@@ -177,6 +177,21 @@ public static function createDatabase($dbName = null)
         Schema::getInstance()->createDatabase($dbName);
     }
 
+    /**
+     * Returns if the given table has an index with the given name
+     *
+     * @param string $table
+     * @param string $indexName
+     *
+     * @return bool
+     * @throws Exception
+     */
+    public static function tableHasIndex($table, $indexName)
+    {
+        $result = Db::get()->fetchOne('SHOW INDEX FROM '.$table.' WHERE Key_name = ?', [$indexName]);
```

Comment:
was thinking could maybe catch an exception here but this would just hide errors I suppose. So be good to keep it for now like this I reckon